crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -46,9 +51,111 @@ impl Query {
     }
 
     // availableRange - pending impl. on IndexerV2
-    // dryRunTransactionBlock
     // coinMetadata
 
+    /// Simulate running a transaction to inspect its effects without
+    /// committing to them on-chain.
+    ///
+    /// `txBytes` either a `TransactionData` struct or a `TransactionKind`
+    ///     struct, BCS-encoded and then Base64-encoded.  The expected
+    ///     type is controlled by the presence or absence of `txMeta`: If
+    ///     present, `txBytes` is assumed to be a `TransactionKind`, if
+    ///     absent, then `TransactionData`.
+    ///
+    /// `txMeta` the data that is missing from a `TransactionKind` to make
+    ///     a `TransactionData` (sender address and gas information).  All
+    ///     its fields are nullable: `sender` defaults to `0x0`, if
+    ///     `gasObjects` is not present, or is an empty list, it is
+    ///     substituted with a mock Coin object, and `gasPrice` defaults to
+    ///     the reference gas price.
+    ///
+    /// `skipChecks` optional flag to disable the usual verification
+    ///     checks that prevent access to objects that are owned by
+    ///     addresses other than the sender, and calling non-public,
+    ///     non-entry functions.  Defaults to false.
+    ///
+    /// `epoch` the epoch to simulate executing the transaction in.
+    ///     Defaults to the current epoch.
+    async fn dry_run_transaction_block(
+        &self,
+        ctx: &Context<'_>,
+        tx_bytes: String,
+        tx_meta: Option<TransactionMeta>,
+        // TODO: this implies existing of `txMeta`
+        skip_checks: Option<bool>,
+        // TODO: why is this u64 but txMeta::gasPrice is BigInt?
+        epoch: Option<u64>,
+    ) -> Result<DryRunResult> {
+        // TODO: how do we want to enforce `tx_meta` and `skip_checks` being (un)set together?
+        if tx_meta.is_some() {
+            match skip_checks {
+                Some(true) => {}
+                _ => {
+                    return Err(Error::Client(
+                        "`skipChecks` must be set to true when `txMeta` is set".to_string(),
+                    ))
+                    .extend();
+                }
+            }
+        }
+        // TODO: whats the point of this
+        let skip_checks = skip_checks.unwrap_or(false);
+
+        let sui_sdk_client: &Option<SuiClient> = ctx
+            .data()
+            .map_err(|_| Error::Internal("Unable to fetch Sui SDK client".to_string()))
+            .extend()?;
+        let sui_sdk_client = sui_sdk_client
+            .as_ref()
+            .ok_or_else(|| Error::Internal("Sui SDK client not initialized".to_string()))
+            .extend()?;
+
+        if let Some(TransactionMeta {
+            sender,
+            gas_price,
+            gas_objects, // TODO: None of the SDK functions use this. Gas us auto created.
```

Comment:
Gas is not always auto-created for dry-run (although it can be) only for dev-inspect, so this exists for the purposes of dry-running a transaction kind.  It does also beg the question of what happens when you try and dev inspect a transaction that has a gas object.  For now, we can just call that out as something watch out for (gas object will be ignored in that case), but eventually (once we move away from delegating all these calls to a fullnode) we will want to be able to handle this case too (dev inspect a transaction that has a gas object predetermined).  Reason being that PTBs that use the gas coin can have their behaviour affected by having the gas object substituted.